### PR TITLE
Fix mac rootdict

### PR DESCRIPTION
**Pull request by gaede**
BEGINRELEASENOTES
- make compatible w/ MacOS:
       -  move all virtual functions to header of ROOTReader
       -  add ROOT libs and header to TestDataModel

ENDRELEASENOTES

### CODE REVIEW

File: tests/CMakeLists.txt [hunk 1]
```diff
@@ -11,11 +11,13 @@ file(GLOB sources_utils utilities/*.cc)
 file(GLOB headers_utils utilities/*.h)
 
 add_library(TestDataModel SHARED ${sources} ${sources_utils} ${headers} ${headers_utils})
-target_link_libraries(TestDataModel podio::podio)
+target_link_libraries(TestDataModel podio::podio ROOT::Core)
 target_include_directories(TestDataModel PUBLIC
   ${CMAKE_SOURCE_DIR}/include
-  ${CMAKE_CURRENT_SOURCE_DIR}/
+  ${CMAKE_CURRENT_SOURCE_DIR}
+  ${ROOT_INCLUDE_DIRS}
```
Comment (andresailer) [inline tests/CMakeLists.txt]: Was this really necessary? ROOT::Core should include this already even on old root versions given this in podio/CMakeLists.txt

https://github.com/AIDASoft/podio/blob/6019f0f3eb9c4924f8fd21d48cbbd30aab8784cc/CMakeLists.txt#L63-L65

Comment (gaede) [reply tests/CMakeLists.txt]: Yes. Without this fix I got:
```
[ 15%] Building CXX object tests/CMakeFiles/TestDataModel.dir/TestDataModel.cxx.o
cd /Users/gaede/podio/podio/build/tests && /Library/Developer/CommandLineTools/usr/bin/c++  -DTestDataModel_EXPORTS -I/Users/gaede/podio/podio/include -I/Users/gaede/podio/podio/tests  -std=c++14 -stdlib=libc++ -O2 -g -DNDEBUG -fPIC   -std=c++14 -o CMakeFiles/TestDataModel.dir/TestDataModel.cxx.o -c /Users/gaede/podio/podio/build/tests/TestDataModel.cxx
/Users/gaede/podio/podio/build/tests/TestDataModel.cxx:13:10: fatal error: 'RConfig.h' file not found
#include "RConfig.h"
         ^~~~~~~~~~~
1 error generated.
```

Comment (gaede) [reply tests/CMakeLists.txt]: ... maybe, because I fixed this first and only then added ROOT:Core ?

Comment (andresailer) [reply tests/CMakeLists.txt]: Probably

Comment (gaede) [reply tests/CMakeLists.txt]: Indeed - see my last commit.

**Comment by gaede** [comment]
Superseded by #81.

**Pull request closed without merging**